apps/alchemist/test/unit/alchemist/broadway_test.exs

```diff
@@ -0,0 +1,120 @@
+defmodule Alchemist.BroadwayTest do
+  use ExUnit.Case
+  use Placebo
+
+  alias SmartCity.TestDataGenerator, as: TDG
+  alias SmartCity.Data
+
+  import SmartCity.Data, only: [end_of_data: 0]
+  import SmartCity.Event, only: [data_standardization_end: 0]
+  import SmartCity.TestHelper, only: [eventually: 1]
+
+  @ingestion_id "ingestion1"
+  @dataset_id "ds1"
+  @topic "raw-ds1"
+  @producer :ds1_producer
+  @current_time "2019-07-17T14:45:06.123456Z"
+  @instance_name Alchemist.instance_name()
+
+  setup do
+    allow Elsa.produce(any(), any(), any(), any()), return: :ok
+    allow SmartCity.Data.Timing.current_time(), return: @current_time, meck_options: [:passthrough]
+
+    ingestion = TDG.create_ingestion(%{id: @ingestion_id})
+
+    {:ok, broadway} =
+      Alchemist.Broadway.start_link(
+        ingestion: ingestion,
+        output: [
+          topic: :output_topic,
+          connection: @producer
+        ],
+        input: [
+          topics: [@topic]
+        ]
+      )
+
+    on_exit(fn ->
+      ref = Process.monitor(broadway)
+      Process.exit(broadway, :normal)
+      assert_receive {:DOWN, ^ref, _, _, _}, 2_000
+    end)
+
+    [broadway: broadway]
+  end
+
+  test "should call Transformers.NoOp with the provided data and return its result", %{broadway: broadway} do
+    data = TDG.create_data(dataset_id: @dataset_id, payload: %{"name" => "johnny", "age" => "21"})
+    kafka_message = %{value: Jason.encode!(data)}
+
+    allow Transformers.NoOp.transform!(kafka_message), return: kafka_message
```

Comment:
Does this need to be mocked?

Reply:
Doesn't need to be I don't think. Thought since it was a unit test it should be mocked.

Happy to do either, not super familiar with trends for that in the other apps.

Reply:
Mocking is generally used in these tests to prevent dependencies on outside processes like kafka, redis, etc. The transformers though are just code. No reason to not let them run.